src/Microsoft.Data/BinaryOperations/DataFrameBinaryOperations.tt

```diff
@@ -0,0 +1,69 @@
+﻿<#@ template debug="false" hostspecific="false" language="C#" #>
+<#@ assembly name="System.Core" #>
+<#@ import namespace="System.Linq" #>
+<#@ import namespace="System.Text" #>
+<#@ import namespace="System.Collections.Generic" #>
+<#@ output extension=".cs" #>
+<#@ include file="..\DataFrameColumnArithmeticTemplate.ttinclude" #>
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.Data
+{
+    public partial class DataFrame
+    {
+        #region Binary Operations
+
+<# foreach (MethodConfiguration method in methodConfiguration) { #>
+<# if (method.MethodType == MethodType.BinaryScalar || method.MethodType == MethodType.ComparisonScalar) {#>
+        public DataFrame <#=method.MethodName#><T>(T value)
+<# } else if (method.MethodType == MethodType.BinaryInt) { #>
+        public DataFrame <#=method.MethodName#><T>(int value)
+<# } else { #>
+        public DataFrame <#=method.MethodName#><T>(IList<T> values)
+<# } #>
+            where T : struct
```

Comment:
Because we want this to be a DataFrame of primitive types. Since there's no way at the moment to specify primitive types alone, I put in value types. A DataFrame is used for data wrangling with primitive types, similar to Pandas. Is there a way to add reference types to Pandas/DataFrame that makes sense? For ex, what does it mean to say DataFrameColumn<someClass>.Max() or DataFrameColumn<someClass>.Add(5)?

Reply:
you should be able to store strings in a dataframe column.

Reply:
I agree. At the moment, my plan is to store strings as `byte`s so I can still keep the struct constraint

Reply:
I filed an issue to supports string columns in the DataFrame. Working on a patch for it at the moment